### PR TITLE
arch/sim: Move spiflash simulation to drivers/spi instead

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
since it's common implementation can be used in other arch too:

- arch/sim: Move spiflash simulation to drivers/spi instead 
- arch/sim: Move qspiflash simulation to drivers/spi instead

## Impact
Minor, reorganize the layout

## Testing
Pass CI

### CODE REVIEW

**Review by acassis — APPROVED** [review]

File: arch/sim/src/sim/up_initialize.c [hunk 1]
```diff
@@ -125,12 +128,12 @@ static void up_init_smartfs(void)
         }
     }
 #endif
-#endif      /* CONFIG_SIM_SPIFLASH */
+#endif      /* CONFIG_SPI_FLASH */
```
Comment (pkarashchenko) [inline arch/sim/src/sim/up_initialize.c:131]:
```suggestion
#endif /* CONFIG_SPI_FLASH */
```

File: drivers/spi/qspi_flash.c [hunk 2]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
 
 /* QSPI methods */
 
-static int         qspiflash_lock(FAR struct qspi_dev_s *dev, bool lock);
-static uint32_t    qspiflash_setfrequency(FAR struct qspi_dev_s *dev,
+static int         qspi_flash_lock(FAR struct qspi_dev_s *dev, bool lock);
+static uint32_t    qspi_flash_setfrequency(FAR struct qspi_dev_s *dev,
                      uint32_t frequency);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:176]:
```suggestion
                                           uint32_t frequency);
```

File: drivers/spi/qspi_flash.c [hunk 3]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
-static void        qspiflash_setmode(FAR struct qspi_dev_s *dev,
+static void        qspi_flash_setmode(FAR struct qspi_dev_s *dev,
                      enum qspi_mode_e mode);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:178]:
```suggestion
                                      enum qspi_mode_e mode);
```

File: drivers/spi/qspi_flash.c [hunk 4]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
-static void        qspiflash_setbits(FAR struct qspi_dev_s *dev, int nbits);
-static int         qspiflash_command(FAR struct qspi_dev_s *dev,
+static void        qspi_flash_setbits(FAR struct qspi_dev_s *dev, int nbits);
+static int         qspi_flash_command(FAR struct qspi_dev_s *dev,
                      FAR struct qspi_cmdinfo_s *cmd);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:181]:
```suggestion
                                      FAR struct qspi_cmdinfo_s *cmd);
```

File: drivers/spi/qspi_flash.c [hunk 5]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
-static int         qspiflash_memory(FAR struct qspi_dev_s *dev,
+static int         qspi_flash_memory(FAR struct qspi_dev_s *dev,
                      FAR struct qspi_meminfo_s *mem);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:183]:
```suggestion
                                     FAR struct qspi_meminfo_s *mem);
```

File: drivers/spi/qspi_flash.c [hunk 6]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
-static FAR void   *qspiflash_alloc(FAR struct qspi_dev_s *dev,
+static FAR void   *qspi_flash_alloc(FAR struct qspi_dev_s *dev,
                                    size_t buflen);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:185]:
```suggestion
                                    size_t buflen);
```

File: drivers/spi/qspi_flash.c [hunk 7]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
-static void        qspiflash_free(FAR struct qspi_dev_s *dev,
+static void        qspi_flash_free(FAR struct qspi_dev_s *dev,
                                   FAR void *buffer);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:187]:
```suggestion
                                   FAR void *buffer);
```

File: drivers/spi/qspi_flash.c [hunk 8]
```diff
@@ -178,22 +171,22 @@ struct sim_qspiflashdev_s
...
 
-static void qspiflash_writeword(FAR struct sim_qspiflashdev_s *priv,
+static void qspi_flash_writeword(FAR struct qspi_flashdev_s *priv,
                     uint16_t data, FAR struct qspi_cmdinfo_s *cmdinfo);
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:190]:
```suggestion
                                 uint16_t data,
                                 FAR struct qspi_cmdinfo_s *cmdinfo);
```

File: drivers/spi/qspi_flash.c [hunk 9]
```diff
@@ -532,23 +525,23 @@ static int qspiflash_command(FAR struct qspi_dev_s *dev,
...
+              p_buf[2] = CONFIG_QSPI_FLASH_CAPACITY;
               break;
 
-            case QSPIFLASH_RDSR:
+            case QSPI_FLASH_RDSR:
               if (priv->wren == 1)
-                  p_buf[0] = QSPIFLASH_WREN_SET;
+                  p_buf[0] = QSPI_FLASH_WREN_SET;
               else
                   p_buf[0] = 0;
```
Comment (pkarashchenko) [inline drivers/spi/qspi_flash.c:546]:
```suggestion
              p_buf[0] = (priv->wren == 1) ? QSPI_FLASH_WREN_SET : 0;
```

**Review by pkarashchenko — APPROVED** [review]

File: arch/sim/src/sim/up_initialize.c (hunk 1 above)
Comment (xiaoxiang781216) [reply arch/sim/src/sim/up_initialize.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 2 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 4 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 5 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 7 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 9 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 6 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 8 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

File: drivers/spi/qspi_flash.c (hunk 3 above)
Comment (xiaoxiang781216) [reply drivers/spi/qspi_flash.c]: Done.

**Comment by xiaoxiang781216** [comment]
@pkarashchenko the new patch is used to verify the change work.

**Review by pkarashchenko — APPROVED** [review]
LGTM!

File: drivers/spi/spi_flash.c [hunk 10]
```diff
@@ -642,15 +635,15 @@ static void spiflash_sectorerase(FAR struct sim_spiflashdev_s *priv)
...
+          len = CONFIG_SPI_FLASH_SECTORSIZE;
         }
-      else if (priv->last_cmd == SPIFLASH_SSE)
+      else if (priv->last_cmd == SPI_FLASH_SSE)
         {
-          address &= CONFIG_SIM_SPIFLASH_SUBSECTORSIZE_MASK;
-          len = CONFIG_SIM_SPIFLASH_SUBSECTORSIZE;
+          address &= CONFIG_SPI_FLASH_SUBSECTORSIZE_MASK;
+          len = CONFIG_SPI_FLASH_SUBSECTORSIZE;
         }
```
Comment (pkarashchenko) [inline drivers/spi/spi_flash.c:647]:
```suggestion
        }
      else
        {
          len = 0;
        }

        if (len > 0)
          {
             /*​ Now perform the erase ​*/ 
  
 ​     ​       memset​(&priv->​data​[address], ​0xff​, len);
          }
```

Comment (xiaoxiang781216) [reply drivers/spi/spi_flash.c:647]: Done, but I skip the check before memset since memset will take care this special case.

**Pull request merged**